CodeModel/src/main/java/org/openzen/zenscript/codemodel/compilation/ResolvedType.java

```diff
@@ -27,13 +28,19 @@ public interface ResolvedType {
 	default Optional<Expression> tryCastExplicit(TypeID target, ExpressionCompiler compiler, CodePosition position, Expression value, boolean optional) {
 		return findCaster(target)
 				.filter(caster -> !caster.getModifiers().isImplicit())
-				.map(caster -> caster.call(compiler.at(position), value, CallArguments.EMPTY));
+				.flatMap(caster -> MatchedCallArguments.match(compiler, position, Collections.singletonList(caster), target, TypeID.NONE)
```

Comment:
These changes were done since we need the ExpansionTypeParameters even for a caster.
Is there a better way than matching the method like this or is this fine?